disk-mapper/internal/setup/setup.go

```diff
@@ -165,6 +167,49 @@ func (s *Manager) saveConfiguration(passphrase []byte) error {
 	return s.config.Generate(stateDiskMappedName, s.diskPath, filepath.Join(keyPath, keyFile), cryptsetupOptions)
 }
 
+// LogDevices logs all available block devices and partitions (lsblk like).
+func (s *Manager) LogDevices() {
+	var devices []fs.FileInfo
+	dirs, err := os.ReadDir("/sys/class/block")
+	for _, file := range dirs {
+		if file.IsDir() {
+			continue
+		}
+		fileInfo, err := file.Info()
+		if err != nil {
+			panic(err)
+		}
+		devices = append(devices, fileInfo)
+	}
+	if err != nil {
+		panic(err)
+	}
```

Comment:
```suggestion
	dirs, err := os.ReadDir("/sys/class/block")
	if err != nil {
		return err
	}
	for _, file := range dirs {
		if file.IsDir() {
			continue
		}
		fileInfo, err := file.Info()
		if err != nil {
			return err
		}
		devices = append(devices, fileInfo)
	}
```
Error should be handled before checking the return value.
Also, this function should not panic, but return errors to the caller